protocol/src/main/protobuf/UserBitShared.proto

```diff
@@ -148,6 +148,8 @@ message SerializedField {
   optional int32 value_count = 4;
   optional int32 var_byte_length = 5;
   optional int32 buffer_length = 7;
+  optional bool is_dup = 8;
+  optional int32 logical_value_count = 9;
```

Comment:
Just had an offline conversation with Paul and Kunal. The agreement is that we should introduce a versioning mechanism at the protocol level so that clients could advertise their version identifier; this way the server can use this information to turn on / off features based on the client capabilities. The task is only to create the preliminary versioning infrastructure; more sophistication can be added later on.